src/Honeycomb.OpenTelemetry/HoneycombOptions.cs

```diff
@@ -76,6 +50,15 @@ static HoneycombOptions()
         /// </summary>
         public string ApiKey { get; set; }
 
+        /// <summary>
+        /// Confirm whether api key is legacy
+        /// </summary>
+        public Boolean isLegacyKey()
+        {
+            // legacy key has 32 characters
+            return ApiKey.Length == 32;

```

Comment:
Need to guard against ApiKey being null.
```suggestion
            return ApiKey?.Length == 32 : true;
```

Reply:
IsLegacyKey() is only ever called after an initial check for API Key, so it should never be null in this check. I adjusted it to return `ApiKey?.Length == 32`, does that work? I can't add `: true` here.

Reply:
Yeah, I wasn't certain I needed the `: true` for the null coalesce. I think it just returns `false` if ApiKey is null, which is fine as you say because it's checked beforehand.